web/src/settings_users.js

```diff
@@ -633,3 +643,44 @@ export function set_up_bots() {
         settings_bots.add_a_new_bot();
     });
 }
+
+export async function fetch_presence_user_setting() {
+    if (page_params.is_spectator) {
+        return;
+    }
+
+    await new Promise((resolve) => {
+        channel.post({
+            url: "/json/users/me/presence",
+            data: {
+                status: compute_active_status(),
+                ping_only: false,
+                last_update_id: -1,
+                history_limit_days: 9999,
+            },
+            success(response) {
+                try {
+                    const data = post_presence_response_schema.parse(response);
+
+                    if (data.presences) {
+                        const updated_presence_info = {};
+
+                        for (const [user_id, presence_data] of Object.entries(data.presences)) {
+                            updated_presence_info[user_id] = {
+                                active_timestamp: presence_data.active_timestamp,
+                                idle_timestamp: presence_data.idle_timestamp,
+                            };
+                        }
+                        Object.assign(user_setting_presence_info, updated_presence_info);
+                    }
+                    resolve();
+                } catch (error) {
+                    throw new Error("Failed to fetch presence data: " + error);
+                }
+            },
+            error(error) {
+                throw new Error("Failed to fetch presence data: " + error);
```

Comment:
Do we really want to throw an error here? I think we should just display users using the information we have instead.

Reply:
Should I use redraw here or just remove this error throwing part from here?